benchmark/bench.js

```diff
@@ -0,0 +1,221 @@
+			continue;
+		}
+		console.info(`\n⌛ repository: ${name}`);
+		const projectDirectory = cloneProject(name, config.repository, config.formattedDirs);
+		console.info("");
+
+		const dirs = config.lintedDirs.map((dir) => `"${dir}"`).join(" ");
+		const biomeCmd = `"${biomeBin}" lint --config-path=../../ --max-diagnostics=0 ${dirs}`;
+		const benchCommands = {
+			eslint: `../../node_modules/.bin/eslint --quiet --config=../../eslint.config.js --no-ignore ${dirs}`,
+			"ts-eslint": `../../node_modules/.bin/eslint --quiet --config=../../ts-eslint.config.js --no-ignore ${dirs}`,
+			biome: biomeCmd,
+			"biome-1-thread": withEnvVariable("RAYON_NUM_THREADS", "1", biomeCmd),
+			"ts-biome": `"${biomeBin}" lint --config-path=../../ts-biome.json --max-diagnostics=0 ${dirs}`,
+		};
+
+		let suite = "";
+		for (const benchName of Object.keys(benchCommands)) {
+			if ("bench" in options && !options.bench.includes(benchName)) {
+				continue;
+			}
+			suite += ` -n "${benchName}" '${benchCommands[benchName]}'`;
+		}
+		if (suite.length === 0) {
+			console.error(`Benchmark '${options.bench}' doesn't exist.`);
+			process.exit(1);
+		}
+
+		let hyperfineCommand = `hyperfine --ignore-failure --shell=${shellOption()} ${suite}`;
+		if (options.verbose) {
+			hyperfineCommand += " --show-output";
+			console.info(`${hyperfineCommand}\n`)
+		}
+
+		execSync(hyperfineCommand, {
+			cwd: projectDirectory,
+			stdio: "inherit",
+		});
+	}
+}
+
+function shellOption() {
+	if (process.platform == "win32") {
+		// Use Powershell so that it is possible to set an environment variable for a single command (ugh!)
+		return "powershell";
+	} else {
+		return "default";
+	}
+}
+
+function withEnvVariable(name, value, command) {
+	switch (process.platform) {
+		case "win32": {
+			return `$Env:${name}=${value}; ${command}`;
+		}
+		default:
+			return `${name}="${value}" ${command}`;
+	}
+}
+
+function withDirectory(cwd) {
+	return {
+		run(command, options) {
+			execSync(command, {
+				cwd,
+				...options,
+			});
+		},
+	};
+}
+
+function cloneProject(name, repository, dirs = []) {
+	const projectDirectory = path.join(TMP_DIR, name);
+
+	const inProjectDirectory = withDirectory(projectDirectory);
+
+	if (fs.existsSync(projectDirectory)) {
+		inProjectDirectory.run("git reset --hard @{u}");
+		inProjectDirectory.run("git clean -df");
+		inProjectDirectory.run("git pull --quiet --depth=1 --ff-only");
+	} else {
+		withDirectory(TMP_DIR).run(
+			`git clone --quiet --depth=1 ${dirs.length > 0 ? "--sparse" : ""} ${repository}`,
+			{
+				stdio: "inherit",
+			},
+		);
+	}
+
+	if (dirs.length > 0) {
+		console.info(
+			`Adding directories ${dirs.join()} to sparse checkout in ${projectDirectory}`,
+		);
+		inProjectDirectory.run(`git sparse-checkout add ${dirs.join(" ")}`);
+	}
+
+	return projectDirectory;
+}
+
+function buildBiome() {
+	console.info("Building Biome...");
+	execSync("cargo build --bin biome --release", {
+		stdio: "inherit",
+	});
```

Comment:
You have to pass the `BIOME_VERSION` environment variable so nursery rules are automatically turned off . As long as it's not `0.0.0`, it should be fine

Reply:
`0.0.0` should be fine?

We don't need this because we turn off `recommended` in the config. However, I am not opposed to make the change.